src/parsley/field.js

```diff
@@ -26,24 +26,36 @@ define('parsley/field', [
     this._bindConstraints();
   };
 
+  var statusMapping = { pending: null, resolved: true, rejected: false };
+
   ParsleyField.prototype = {
     // # Public API
     // Validate field and trigger some events for mainly `ParsleyUI`
-    // @returns validationResult:
-    //  - `true` if field valid
-    //  - `[Violation, [Violation...]]` if there were validation errors
+    // @returns `true`, an array of the validators that failed, or
+    // `null` if validation is not finished. Prefer using whenValidate
     validate: function (force) {
+      var promise = this.whenValidate(force);
+      switch (promise.state()) {
+        case 'pending': return null;
+        case 'resolved': return true;
+        case 'rejected': return this.validationResult;
+      };
+    },
+
+    // Validate field and trigger some events for mainly `ParsleyUI`
+    // @returns a promise that succeeds only when all validations do.
+    whenValidate: function (force) {
+      var that = this;
+
       this.value = this.getValue();
 
       // Field Validate event. `this.value` could be altered for custom needs
       this._trigger('validate');
 
-      this._trigger(this.isValid(force, this.value) ? 'success' : 'error');
-
-      // Field validated event. `this.validationResult` could be altered for custom needs too
-      this._trigger('validated');
-
-      return this.validationResult;
+      return this.whenValid(force, this.value)
+      .done(function()   { that._trigger('success'); })
```

Comment:
Personal taste: when I use promises I usually increment these methods to show more distincly they results to action above. Find that more readable, but again just personal taste.